include/internal/catch_test_spec_parser.cpp

```diff
@@ -20,8 +20,12 @@ namespace Catch {
         m_substring.reserve(m_arg.size());
         m_patternName.reserve(m_arg.size());
         m_realPatternPos = 0;
-        for( m_pos = 0; m_pos < m_arg.size(); ++m_pos )
-            visitChar( m_arg[m_pos] );
+        try{
+            for( m_pos = 0; m_pos < m_arg.size(); ++m_pos )
+              visitChar( m_arg[m_pos] );
+        }catch(...){
```

Comment:
technically here you should use `CATCH_TRY` and `CATCH_CATCH_ALL` to avoid problems with disabled exceptions.
But with disabled exceptions `CATCH_CATCH_ALL` expands to `if ((false))` so that would not behave as you want.
Would it be possible to change `visitchar` to return something what would indicate error?

Reply:
Yes, I thought about it, perhaps this is what I'll do